firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -103,4 +124,105 @@ String getApplicationId() {
   String getName() {
     return firebaseApp.getName();
   }
+
+  /**
+   * Create firebase installation id for the {@link FirebaseApp} on FIS Servers and client side
+   * cache.
+   *
+   * <pre>
+   *     The workflow is:
+   *         check if cache empty or cache status is REGISTER_ERROR
```

Comment:
This description is somewhat confusing to me: What happens if the status is REGISTER_ERROR or not?

Reply:
Also what happens if status is UNREGISTERED?

Reply:
Updated the description. Let me know, if you want me to change further.